internal/gke/kubernetes_test.go

```diff
@@ -244,6 +244,97 @@ func TestGetFetchableResourceTypes(t *testing.T) {
 	}
 }
 
+func TestGetResources(t *testing.T) {
+	resourceOne := map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"name": "some-object-one",
+		},
+	}
+	resourceTwo := map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"name": "some-object-two",
+		},
+	}
+	resourceType1 := ResourceType{Group: "apps", Version: "v1", Name: "deployments", Namespaced: true}
+	namespace1 := "my-first-namespace"
+
+	resourceType2 := ResourceType{Group: "apps", Version: "v1", Name: "pods", Namespaced: true}
+	namespace2 := "my-second-namespace"
+
+	dynCliMock := &kubeDynamicClientMock{
+		ResourceFn: func(resource schema.GroupVersionResource) dynamic.NamespaceableResourceInterface {
+			if resource.Group == resourceType1.Group && resource.Version == resourceType1.Version && resource.Resource == resourceType1.Name {
+				return &kubeNamespacedResourceMock{
+					NamespaceFn: func(n string) dynamic.ResourceInterface {
+						if n == namespace1 {
+							return &kubeNamespacedResourceMock{
```

Comment:
I suggest extracting creation of a parametrized `kubeNamespacedResourceMock` to separate func as it is repeated four times in a test.